src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -99,9 +99,21 @@ static List<String> getFieldNamesFromProjectStage(BsonDocument projectStage) {
         return fieldNames;
     }
 
-    private static boolean trueOrOne(BsonValue value) {
-        return (value.isBoolean() && value.asBoolean().getValue())
-                || (value.isNumber() && value.asNumber().intValue() == 1);
+    private static boolean isExcludeProjectSpecification(Map.Entry<String, BsonValue> specification) {
+        var key = specification.getKey();
+        var value = specification.getValue();
+        var exclude = (value.isBoolean() && !value.asBoolean().getValue())
+                || (value.isNumber() && value.asNumber().intValue() == 0);
+        if (exclude && !key.equals(ID_FIELD_NAME)) {
+            throw new RuntimeException(format(
+                    "Exclusions are not allowed in `$project` specifications, except for the [%s] field: [%s, %s]",
+                    ID_FIELD_NAME, key, value));
```

Comment:
This is hardly to ever change, unless we start requiring a schema or start inferring it.

Reply:
well, it is native query related. Currently let us keep it in mind and then move on.

Reply:
Could you please elaborate on why this is related to native queries?

Reply:
because for non-native query we have 100% control of the MQL generation, including the `$project` stage. If we saw no need to use `exclusion list` (from my understanding, explicit inclusion list suffices and should be relies upon) during MQL translation, this issue won't bother us at all.

Reply:
That still does not explain your _"well, it is native query related"_ comment. Do you think native queries will have to use exclustions? If they do and we allow that, how can we possibly know the list of the columns for `ResultSet` without inferring that list from the returned documents, which we don't have plans to do?

Reply:
Sorry to jump in here.  The way I understand it, native queries will not _have_ to use exclusions.  The question is whether we _allow_ them to.  Since I also can't see a way to make it work, I don't think we will allow them to.

Reply:
Right. And if we are not going to allow them to use exclusions, then the code this thread is about will likely not change, which is what I said in the first clarifying comment for reviewers. And that makes me wonder, what is it that we need to keep in mind in relation to this code and native queries (I am trying to understand the comment https://github.com/mongodb/mongo-hibernate/pull/72#discussion_r2003415926, assuming it expressed or implied something that I am missing).

Reply:
> That still does not explain your _"well, it is native query related"_ comment. Do you think native queries will have to use exclustions? If they do and we allow that, how can we possibly know the list of the columns for `ResultSet` without inferring that list from the returned documents, which we don't have plans to do?

I didn't imply that. Sorry for misunderstanding. I meant the exclusion list could only happen in theory in native query. Native query has many edge cases and this is another one.